app/src/main/java/com/github/sdp/mediato/CreateProfileFragment.java

```diff
@@ -29,108 +33,151 @@
  */
 public class CreateProfileFragment extends Fragment {
 
+    private class UsernameWatcher implements TextWatcher{
+
+        private TextInputLayout usernameTextInput;
+        private TextInputEditText usernameEditText;
+
+        UsernameWatcher(TextInputLayout usernameTextInput, TextInputEditText usernameEditText) {
+
+            this.usernameTextInput = usernameTextInput;
+            this.usernameEditText = usernameEditText;
+        }
+
+        @Override
+        public void onTextChanged(CharSequence s, int start, int before, int count) {
+            if (UsernameError.GOOD == CreateProfileFragment.this.isUsernameValid(usernameEditText.getText())) {
+                usernameTextInput.setError(null);
+            }
+        }
+
+        // Useless functions
```

Comment:
Then why keep them ?

Reply:
I think they need to be implemented because of the TextWatcher interface.

Reply:
Yeah I have to implement them but I don't need them